src/pelz/main.c

```diff
@@ -55,33 +111,221 @@ int main(int argc, char **argv)
     default:
       return 1;
     }
   }
 
-  if (optind == 1 )
+  if ((argv[arg_index + 1] != NULL) && (memcmp(argv[arg_index + 1], "exit", 4) == 0))
+  {
+    msg = (char *) calloc(8, sizeof(char));
+    memcpy(msg, "pelz -e", 7);
+    pelz_log(LOG_DEBUG, "Message: %s", msg);
+    write_to_pipe(msg);
+    free(msg);
+  }
+  else if ((argv[arg_index + 1] != NULL) && (memcmp(argv[arg_index + 1], "load", 4) == 0))
+  {
+    pelz_log(LOG_DEBUG, "Load option");
+    if ((argv[arg_index + 2] != NULL) && (memcmp(argv[arg_index + 2], "cert", 4) == 0))
+    {
+      pelz_log(LOG_DEBUG, "Load cert option");
+      if (argv[arg_index + 3] != NULL)
+      {
+        pelz_log(LOG_DEBUG, "Load cert <path> option");
+        path_id_size = strlen(argv[arg_index + 3]) + 1;
+        if (path_id_size > 1)
+        {
+          path_id = (char *) malloc(path_id_size * sizeof(char));
+          memcpy(path_id, argv[arg_index + 3], path_id_size);
+        }
+        pelz_log(LOG_DEBUG, "<path> set: %.*s", (int) path_id_size, path_id);
+        msg = (char *) calloc((12 + path_id_size), sizeof(char));
+        memcpy(msg, "pelz -l -c ", 11);
+        memcpy(&msg[11], path_id, path_id_size);
+        pelz_log(LOG_DEBUG, "Message: %s", msg);
+        write_to_pipe(msg);
+        free(msg);
+        free(path_id);
+      }
+      else
+      {
+        load_usage();
+        return 1;
+      }
+    }
+    else if ((argv[arg_index + 2] != NULL) && (memcmp(argv[arg_index + 2], "private", 3) == 0))
+    {
+      pelz_log(LOG_DEBUG, "Load private option");
+      if (argv[arg_index + 3] != NULL)
+      {
+        pelz_log(LOG_DEBUG, "Load private <path> option");
+        path_id_size = strlen(argv[arg_index + 3]) + 1;
+        if (path_id_size > 1)
+        {
+          path_id = (char *) malloc(path_id_size * sizeof(char));
+          memcpy(path_id, argv[arg_index + 3], path_id_size);
+        }
+        pelz_log(LOG_DEBUG, "<path> set: %.*s", (int) path_id_size, path_id);
+        msg = (char *) calloc((12 + path_id_size), sizeof(char));
+        memcpy(msg, "pelz -l -p ", 11);
+        memcpy(&msg[11], path_id, path_id_size);
+        pelz_log(LOG_DEBUG, "Message: %s", msg);
+        write_to_pipe(msg);
+        free(msg);
+        free(path_id);
+      }
+      else
+      {
+        load_usage();
+        return 1;
+      }
+    }
+    else
+    {
+      load_usage();
+      return 1;
+    }
+  }
+  else if ((argv[arg_index + 1] != NULL) && (memcmp(argv[arg_index + 1], "remove", 6) == 0))
+  {
+    pelz_log(LOG_DEBUG, "Remove option");
+    if ((argv[arg_index + 2] != NULL) && (memcmp(argv[arg_index + 2], "key", 3) == 0))
+    {
+      pelz_log(LOG_DEBUG, "Remove key option");
+      if (all && (argv[arg_index + 3] == NULL))
+      {
+        pelz_log(LOG_DEBUG, "Remove key --all option");
+        msg = (char *) calloc(14, sizeof(char));
+        memcpy(msg, "pelz -r -k -a", 13);
+        pelz_log(LOG_DEBUG, "Message: %s", msg);
+	write_to_pipe(msg);
```

Comment:
several lines are not correctly tabbed, this is one example

Reply:
Lines should be fixed with correct spacing.